### DISCUSSION TITLE
[Proposal]: Allow trailing comma in `ComplexElementInitializerExpression`

### DISCUSSION THREAD

**Discussion by Youssef1313** [dotnet/csharplang#6243] [Language Ideas]
# Allow trailing comma in `ComplexElementInitializerExpression`

* [x] Proposed
* [ ] Prototype: Not Started
* [ ] Implementation: Not Started
* [ ] Specification: Not Started

## Summary
[summary]: #summary

Allows a trailing comma in `ComplexElementInitializerExpression`

## Motivation
[motivation]: #motivation

- All initializer expressions currently allows a trailing comma, except `ComplexElementInitializerExpression`.
- Allowing a trailing comma helps for a better diff if a new expression will be added.

## Detailed design
[design]: #detailed-design

Changes to grammar of [draft-v7 branch](https://github.com/dotnet/csharpstandard/blob/draft-v7/standard/grammar.md) is as follows:

```diff
// Source: §11.7.15.4 Collection initializers
collection_initializer
    : '{' element_initializer_list '}'
    | '{' element_initializer_list ',' '}'
    ;

element_initializer_list
    : element_initializer (',' element_initializer)*
    ;

element_initializer
    : non_assignment_expression
    | '{' expression_list '}'
+   | '{' expression_list ',' '}'
    ;

expression_list
    : expression
    | expression_list ',' expression
    ;
```

## Drawbacks
[drawbacks]: #drawbacks

## Alternatives
[alternatives]: #alternatives

## Unresolved questions
[unresolved]: #unresolved-questions

**Comment by Youssef1313** [comment]
An implementation might be as simple as https://github.com/dotnet/roslyn/pull/62140

**Comment by 333fred** [comment]
In previous ldm discussions of https://github.com/dotnet/csharplang/issues/391, we've indicated that we'd like to generally support trailing commas. This would be one more case of it.

**Reply by MgSam** [reply]
Didn't realize the LDT ever discussed. Glad to hear there is support for this!
